apps/ui/src/helpers/turndownService.ts

```diff
@@ -1,28 +1,95 @@
+  linkStyle: 'inlined',
+  linkReferenceStyle: 'full'
 });
 
 export default function (options: { discussion?: string } = {}) {
   turndownService.addRule('table', {
     filter: ['table'],
     replacement: function (content, node) {
-      const headers = Array.from(node.querySelectorAll('thead th')).map(
-        (th: any) => th.textContent
+      // Get headers from thead or first row with th elements
+      let headers = Array.from(node.querySelectorAll('thead th')).map(
+        (th: any) => turndownService.turndown(th.innerHTML).trim()
       );
-      const rows = Array.from(node.querySelectorAll('tbody tr')).map(
-        (tr: any) =>
-          Array.from(tr.querySelectorAll('td')).map((td: any) => td.textContent)
+      let headerCells = Array.from(node.querySelectorAll('thead th'));
+
+      // If no thead, check for th elements in tbody (first row)
+      if (headers.length === 0) {
+        const firstRow = node.querySelector('tbody tr');
+        if (firstRow) {
+          const thElements = firstRow.querySelectorAll('th');
+          if (thElements.length > 0) {
+            headers = Array.from(thElements).map((th: any) =>
+              turndownService.turndown(th.innerHTML).trim()
+            );
+            headerCells = Array.from(thElements);
+          }
+        }
+      }
+
+      // If no headers found, create empty headers based on first row cell count
+      if (headers.length === 0) {
+        const firstDataRow =
+          node.querySelector('tbody tr') || node.querySelector('tr');
+        if (firstDataRow) {
+          const cells = firstDataRow.querySelectorAll('td, th');
+          headers = Array.from(cells).map(() => '');
+          headerCells = Array.from(cells);
+        }
+      }
+
+      // Get alignment from header cells
+      const alignments = headerCells.map((th: any) => {
+        const style = th.style.textAlign || th.getAttribute('align') || '';
+        switch (style.toLowerCase()) {
+          case 'left':
+            return ':---';
+          case 'right':
+            return '---:';
+          case 'center':
+            return ':---:';
+          default:
+            return '---';
+        }
+      });
+
+      // Get data rows, excluding the header row if it's in tbody
+      const bodyRows = Array.from(node.querySelectorAll('tbody tr'));
+      const allRows =
+        bodyRows.length > 0
+          ? bodyRows
+          : Array.from(node.querySelectorAll('tr'));
+      let dataRows = allRows;
+
+      // If headers were found in first tbody row, skip it for data
+      if (headers.length > 0 && allRows.length > 0) {
+        const firstRow = allRows[0] as HTMLTableRowElement;
+        if (firstRow.querySelectorAll('th').length > 0) {
+          dataRows = allRows.slice(1);
+        }
+      }
+
+      const rows = dataRows.map((tr: any) =>
+        Array.from(tr.querySelectorAll('td, th')).map((cell: any) =>
+          turndownService.turndown(cell.innerHTML).trim()
```

Comment:
why trim than before? 

paste:
```md
|  |
| --- |
| HELLO |
```

In preview:

<img width="199" height="140" alt="Image" src="https://github.com/user-attachments/assets/ea91e12f-f9df-44e7-8a50-5b4feecc2012" />

In markdown editor:

<img width="203" height="156" alt="Image" src="https://github.com/user-attachments/assets/4ef5c3b2-c927-4d46-b745-aa55711f6c23" />

Reply:
The behavior from the visual editor is as expected, and similar to other visual editor, like notion.

An empty cell should have the same height as a cell with content, to prevent cell collapsing, and making it hard to select.

After saving, this cell will have the same height as the one in preview.

We could prevent this inconsistency by forcing all cells to have a minimum height, like in visual editor, or keep the inconsistency. But for UX reasons, I will refrain from having the empty cells in visual editor harder to select